pkg/sensors/handler.go

```diff
@@ -89,11 +89,24 @@ func (h *handler) updatePolicyFilter(tp tracingpolicy.TracingPolicy, tpID uint64
 }
 
 func (h *handler) addTracingPolicy(op *tracingPolicyAdd) error {
-	if _, exists := h.collections[op.name]; exists {
+	// allow overriding existing policy collection with a load error as all
+	// sensors were unloaded (if some succeeded to load) and the policy is
+	// disabled
+	if col, exists := h.collections[op.name]; exists && col.loadError == nil {
 		return fmt.Errorf("failed to add tracing policy %s, a sensor collection with the name already exists", op.name)
 	}
 	tpID := h.allocPolicyID()
 
+	col := &collection{
+		name:            op.name,
+		tracingpolicy:   op.tp,
+		tracingpolicyID: uint64(tpID),
+	}
+	// Register the collection early even if there is a potential loading error.
+	// col.loadError should be set before returning the error so that error
+	// state is saved and the policy can be later overridden
+	h.collections[op.name] = col
```

Comment:
There is an interim state here where the new collection is part of .collections but it's state is wrong (appears as if it's properly working). This is fine, because we serialize everything in a single goroutine so the state is not visible.

If we care about it, we could do:
```
defer func() {
    h.collections[op.name] = col
}()
```
An alternative would be to have a state in the collection: {LOADING, RUNNING, ERROR, ...}.

In any case, no need to change anything just sharing some thoughts as I'm reading the code in case we ever want to do something more complicated to allow for concurrent operations since we 've discussed this offline.

Reply:
yep indeed, let's keep writing value instead of pointers at appropriate timing https://github.com/cilium/tetragon/pull/2090#discussion_r1492559364. Let's keep this simple.